cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+                len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0
+            )
+
+        def csv_import(self, vm_list):
+            temp_file = tempfile.NamedTemporaryFile(suffix=".csv")
+            with open(temp_file.name, "w") as file:
+                headers = ["Name", "Provider"]
+                writer = csv.DictWriter(file, fieldnames=headers)
+                writer.writeheader()
+                for vm in vm_list:
+                    writer.writerow({"Name": vm.name, "Provider": vm.provider.name})
+            self.hidden_field.fill(temp_file.name)
+
+        def fill(self, values):
+            csv_import = values.get('csv_import')
+            vm_list = values.get('vm_list')
+            if csv_import:
+                self.csv_import(vm_list)
+            for vm in vm_list:
+                self.filter("VM Name", vm.name)
+                for row in self.table.rows():
+                    if vm.name in row.vm_name.read():
+                        row[0].fill(True)
+                self.clear_filters.click()
+            self.parent.next_btn.click()
+
+        def after_fill(self, was_change):
+            self.browser.send_keys(Keys.ENTER, self.search_box)
+
+        def filter(self, type, name):
+            try:
+                self.filter_by_dropdown.item_select(type)
+            except NoSuchElementException:
+                logger.info("{} not present in filter dropdown!".format(type))
+            self.search_box.fill(name)
+            self.after_fill(was_change=True)
+
+        def select_by_name(self, vm_name):
+            self.filter("VM Name", vm_name)
+            vms_selected = []
+            for row in self.table.rows():
+                if vm_name in row.read()["VM Name"]:
+                    row.select.fill(True)
+                    vms_selected.append(row.read()["VM Name"])
+            return vms_selected
+
+    @View.nested
+    class instance_properties(View):  # noqa
+        table = Table('.//div[contains(@class, "container-fluid")]/table')
+        edit_instance_properties = Text(locator='.//button/span[contains(@class, "pficon-edit")]')
+        select_security_group = Select(locator='.//td[5]/select')
+        select_flavor = Select(locator='.//td[6]/select')
+        save_properties = Text(locator='.//div[contains(@class, "inline-edit-buttons")]'
+                                       '/button[contains(@aria-label, "Save")]')
+
+        @property
+        def is_displayed(self):
+            return (self.table.is_displayed and
+                    (len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0))
+
+        def fill(self, values):
+            osp_security_group = values.get('osp_security_group')
+            osp_flavor = values.get('osp_flavor')
+            if osp_security_group or osp_flavor:
+                self.edit_instance_properties.click()
+                self.select_security_group.wait_displayed()
+                if osp_security_group:
```

Comment:
These checks block the fill from actually being able to select nothing for the security group, is that intentional?

Reply:
Yes , it needs to be filled only for one test case . Added docblock